### PR TITLE
Add BrokenLinkTemplate for broken/errored linksTo fields

### PR DESCRIPTION
**Pull request by habdelra**
## Goal

Build the system-provided visual placeholder dispatched in place of a card when a `linksTo` / `linksToMany` target is in `error` or `not-found` state. Treatment B from the [linked-card placeholder design](https://linear.app/cardstack/issue/CS-11203/markdown-embedded-card-placeholder) — a sized box that matches the original card's footprint.

Tracks [CS-11214](https://linear.app/cardstack/issue/CS-11214/placeholder-visual-component-for-brokenerrored-linksto-fields) (T3 of the broken-linksTo project — independent of the field-side work, buildable + testable in isolation).

## What's here

- New `packages/base/default-templates/broken-link-template.gts` mirroring the shape of `missing-template.gts`. **Not a CardDef** — render-only.
- API exactly as specified in the ticket:
  ```ts
  interface BrokenLinkTemplateArgs {
    brokenUrl: string;
    errorDoc: SerializedError;
    state: 'error' | 'not-found';
    format: 'isolated' | 'fitted' | 'embedded' | 'atom';
  }
  ```
- Per-format sizing matches the parent container defaults in `field-component.gts:450-481`. The fitted variant uses `container-type: size`; embedded uses `container-type: inline-size`; atom is `inline-flex`; isolated fills the content area. Container queries collapse the headline / status when the host slot is < 200px wide or ≤ 65px tall (the `.linksToMany-itemContainer` case).
- **NotFound** emphasises the URL — the `Could not find <url>` message that error docs ship with would be duplicative, so it's suppressed for this state.
- **Error** renders the message, stack, and `additionalErrors` so an AI consumer reading the DOM can recover the underlying failure.
- Component test (`packages/host/tests/integration/components/broken-link-template-test.gts`) covers each format × state combination plus the minimal-`errorDoc` and unsafe-URL edges.
- `realm-indexing-test.gts` manifest snapshot updated to include the new module.

## Preview

Static render of every format × state combination (the page is self-contained — boxel CSS variables are inlined). Linked from a temporary file committed to this branch; a followup commit removes it from HEAD but the raw URL is pinned to the commit SHA so the image survives.

BrokenLinkTemplate preview

A v2 preview reflecting the embedded-format tightening (line-clamped message, smaller font) is in the PR comment thread.

## Test plan

- [ ] CI runs `broken-link-template-test.gts` and `realm-indexing-test.gts`.
- [ ] Visual review of the preview screenshot — does treatment B match Chris's mock?
- [ ] (Followup) Once T14/T15 land, the placeholder gets dispatched live by `field-component`/`links-to-many-component`; that's where end-to-end render-count behaviour will be exercised.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/base/default-templates/broken-link-template.gts [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+        </div>
+        <a
+          class='url'
+          href={{@brokenUrl}}
+          target='_blank'
+          rel='noopener noreferrer'
+          data-test-broken-link-url
+        >
+          {{@brokenUrl}}
+        </a>
```
[the remark this answers is not in the document]
Comment (habdelra) [reply packages/base/default-templates/broken-link-template.gts]: i think there is a sanitizeHTML function that we use from runtime-common that might be helpful here...

[the remark this answers is not in the document]
Comment (habdelra) [reply packages/base/default-templates/broken-link-template.gts]: [Claude Code 🤖] 

Fixed in 97298e4. The href is now gated by `isSafeHttpUrl(brokenUrl)` — parsed via `new URL(...)` and checked against an `http:` / `https:` allowlist. If the protocol is anything else (`javascript:`, `data:`, malformed, empty), the URL renders inside a `<span class="url">` instead of an `<a>` so the value stays visible to a reviewer but clicking it cannot navigate. Added a regression test (`renders unsafe URLs as plain text rather than an anchor`) using `javascript:alert(1)` as the input.

(I also briefly looked at routing this through `sanitizeHtml` from runtime-common, per Hassan's nearby comment, but landed on the explicit protocol check — reasoning in that thread.)

File: packages/base/default-templates/broken-link-template.gts [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+        stack: typeof entry.stack === 'string' ? entry.stack : undefined,
+      });
+    }
+    return normalized;
+  }
+
+  <template>
+    <div
+      class='broken-link-template {{@format}} {{@state}}'
+      role='alert'
```
[the remark this answers is not in the document]
Comment (habdelra) [reply packages/base/default-templates/broken-link-template.gts]: [Claude Code 🤖] 

Agreed — dropped `role='alert'` in 97298e4. A `linksToMany` row with multiple broken slots would have fired multiple assertive announcements per render which is exactly the disruptive pattern you're flagging. Visually-identifying styling (dashed border, red marker, error background tint) still conveys the broken state without relying on an ARIA live region. If we later decide a polite announcement is useful for a single isolated-format failure I'd add `role='status'` only there, not on the placeholder root.

File: packages/base/default-templates/broken-link-template.gts [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+                    {{#if err.stack}}
+                      <pre
+                        class='additional-stack'
+                      >{{err.stack}}</pre>
+                    {{/if}}
+                  </li>
+                {{/each}}
+              </ul>
+            </details>
+          {{/if}}
```
[the remark this answers is not in the document]
Comment (habdelra) [reply packages/base/default-templates/broken-link-template.gts]: [Claude Code 🤖] 

Fixed in 97298e4. Wrapped the additional-errors block in `{{#let this.additionalErrors as |additionalErrors|}}` so the getter is invoked once per render and the normalized array is reused across the `.length` check, the summary text, and the `{{#each}}`. Kept the getter pure (no tracked mutation) so it composes safely with the Glimmer-discipline rules in the broader broken-linksTo plan.

File: packages/host/tests/integration/components/broken-link-template-test.gts [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+        format: 'fitted',
+      });
+
+      assert.dom('[data-test-broken-link-template="fitted"]').exists();
+      assert.dom('[data-test-broken-link-headline]').exists();
+      assert.dom('[data-test-broken-link-url]').hasText(NOT_FOUND_URL);
+      assert
+        .dom('[data-test-broken-link-message]')
+        .doesNotExist('fitted suppresses the prose error message');
+    });
```
[the remark this answers is not in the document]
Comment (habdelra) [reply packages/host/tests/integration/components/broken-link-template-test.gts]: [Claude Code 🤖] 

Fair catch — the PR description claimed full cross-product coverage but the matrix was missing two cells. Added in 97298e4:

- `not-found / embedded` — asserts headline + URL, confirms the "Could not find ..." message stays suppressed and the stack panel is absent.
- `error / fitted` — asserts the fitted variant suppresses the prose message even when there's a real error reason worth surfacing, and that the stack panel stays isolated-only.

Plus the `javascript:` URL regression test added in the same commit, so the suite is now 4 formats × 2 states + edge cases (minimal errorDoc, unsafe URL).

File: packages/base/default-templates/broken-link-template.gts (hunk 1 above)
[the remark this answers is not in the document]
Comment (habdelra) [reply packages/base/default-templates/broken-link-template.gts]: [Claude Code 🤖] 

I did look at `sanitizeHtml` from `runtime-common/dompurify-runtime.ts` — it runs DOMPurify on an HTML *string* and returns the cleaned HTML, intended for cases like markdown render output or AI-generated body content. Two reasons it didn't quite fit here:

1. **The XSS surface is the `href` attribute binding, not an HTML string.** Glimmer auto-escapes the link text (`{{@brokenUrl}}`) so there's no innerHTML injection risk; the only dangerous path is `href={{...}}` evaluating a `javascript:` URL on click. To use `sanitizeHtml` I'd have to construct the anchor as an HTML string, sanitize it, and pipe through `htmlSafe(...)` into the template — losing Glimmer's attribute binding for a single attribute value.

2. **Scoped CSS would break.** The template's class names (`url`, `marker`, etc.) get compile-time-suffixed by Glimmer's scoped-CSS pass. A hand-written `<a class="url">` inside an `htmlSafe` string wouldn't get that suffix and would silently lose its styling.

So I went with an explicit protocol allowlist via `new URL(...)`, falling back to plain-text rendering on `javascript:` / `data:` / malformed input (commit 97298e4, regression test added). That's functionally equivalent to DOMPurify's built-in URL-attribute sanitization but stays on the Glimmer-idiomatic attribute-binding path.

Happy to switch to the `sanitizeHtml` + `htmlSafe` approach if you'd rather have the chokepoint live there — let me know and I'll redo it.

**Comment by habdelra** [comment]
[Claude Code 🤖] 

Updated preview after the embedded-overflow fix in cb3bb2d:

BrokenLinkTemplate preview v2

What changed visually since the v1 screenshot:

- **embedded · error** no longer overflows its slot. Tightened the embedded format to `font: var(--boxel-font-xs)` + `gap: 2px` + `padding: var(--boxel-sp-xs)`, and clamped the prose error message to two lines via `-webkit-line-clamp: 2` so the placeholder fits even when the parent slot is height-constrained (e.g. the 110px-tall container the v1 preview used). The full message stays in the DOM for screen readers / AI consumers — only the visual rendering is clipped.
- **embedded · not-found** is now slightly more compact too (same padding/font tightening), but content was already short so no truncation kicks in.
- All other formats (isolated / fitted / atom / linksToMany row) are unchanged — they were already fitting.

**Review by backspace — APPROVED** [review]
In the preview image the atom versions look a couple of pixels too low for me, but looks good otherwise!
